components/needs_team/needs_team.jsx

```diff
@@ -160,15 +149,8 @@ export default class NeedsTeam extends React.Component {
         document.addEventListener('keydown', this.shortcutKeyDown);
     }
 
-    componentDidUpdate(prevProps, prevState) {
-        if (!Utils.areObjectsEqual(prevState.theme, this.state.theme)) {
-            Utils.applyTheme(this.state.theme);
```

Comment:
Don't we still need this if the user changes their theme while looking at the app or if they change it from another tab?

Reply:
We don't need `componentDidUpdate` but your comment prompted me to add it to `componentWillReceiveProps` instead.

Reply:
I'd prefer to have it in componentDidUpdate (or maybe componentWillUpdate, but we don't generally use that one). Generally, we only use componentWillReceiveProps to do state changes based on the props changing.

Reply:
Yes `theme` is now found in props and isn't present in state.

Reply:
I mean that we usually only use `componentWillReceiveProps` to call `this.setState`. Other changes like this are usually handled by `componentDidUpdate`

Reply:
@hmhealey Are you proposing a change in which React lifecycle method we use here? I'm not sure I follow.

Reply:
@mkraft Yes, I'm thinking that we should move the `Utils.applyTheme` back to use `componentDidUpdate` like we were before because we only use `componentWillReceiveProps` to update this.state in other components

Reply:
@hmhealey I see your point about sticking to a convention. The counter-point is that from an efficiency point of view [`this.setState()` generally doesn’t trigger `componentWillReceiveProps()`](https://reactjs.org/docs/react-component.html#componentwillreceiveprops) so the code
```javascript
if (!Utils.areObjectsEqual(nextProps.theme, this.props.theme))
```
 won't need to be evaluated as frequently. Perhaps in cases where state is not used we should change our convention to use `componentWillReceiveProps`.

Reply:
That's a good point that it should be slightly more efficient, but I think any gains would be negligible, so I'd rather stick with convention for the time being, even if it's not the best. As it is, the only thing remaining in state is the current team (which really should be passed in through props, but that's beside the point), so it'll still be rare that a state change triggers this when it doesn't need to. Also, since `getTheme` is a selector, you can also change the check to be `if (nextProps.theme !== this.props.theme)` to make it virtually free

I'd be open to changing our convention here, but unless we make the decision to do it everywhere, I'd rather we keep this code working the same way.